### ISSUE TITLE
icinga2 Provider, Host resource bug

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-icinga2#1]
_This issue was originally opened by @robrankin as hashicorp/terraform#14819. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

The Icinga2 Host resource creates an Icinga2 Host object as follows:

```
object Host "host" {
	action_url = ""
	address = "192.168.1.1"
	address6 = ""
	check_command = "hostalive"
	display_name = ""
	groups = null
	notes = ""
	notes_url = ""
	templates = null
	vars = {
		something = "blart"
	}
	version = 1495628446.109900
	zone = "master"
}
```

The `groups = null` syntax does not appear to be valid.  When attempting to use Icinga2's group assign functionality, `assign where`, to assign the above Host object to Hostgroup object, the `groups = null` entry causes Icinga2 to segfault.

As far as I can tell so far it's only acceptable for the `groups` parameter to have a non-empty list of 1 or more members.

`groups = ["",]` does not work for example.

Icinga2 bug report here: [Icinga2 Issue hashicorp/terraform#5288](https://github.com/Icinga/icinga2/issues/5288)

### Terraform Version
`Terraform v0.9.4`

### Affected Resource(s)
Please list the resources as a list, for example:
- icinga2_host

### Terraform Configuration Files
```hcl
resource "icinga2_host" "host" {
  hostname      = "${var.hostname}"
  address       = "${var.address}"
  check_command = "${var.checkCommand}"

  vars {
    something = "blart"
  }
}
```

### Expected Behavior
The Icinga2 Host resource creates correct Icinga2 object syntax

### Actual Behavior
The Icinga2 Host resource creates incorrect Icinga2 object syntax

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### References
https://github.com/Icinga/icinga2/issues/5288

**Comment by lrsmith** [comment]
This has been vendor'd out to a release which did not require it. Also https://github.com/lrsmith/go-icinga2-api has been modified and versioned as v0.2.x to support not requiring having groups.

**Comment by rayterrill** [comment]
@lrsmith I'm running into the groups = null error when trying to create hosts in Icinga. I see it looks like you fixed in the underlying go-icinga2-api library - did this not get released yet in the provider?

**Comment by rayterrill** [comment]
@lrsmith Any word on this one mate?

**Comment by lrsmith** [comment]
@rayterrill This has been merged into master but a release has not cut yet for it. I hope to cut a release soon with all the recent PRs.

**Comment by jc1arke** [comment]
@lrsmith any news on a new release?

**Comment by lrsmith** [comment]
@jc1arke @rayterrill I am working on getting a new release. Right now the build pipeline is failing and I'm don't have the needed permissions to make the needed changes and I'm working to get the needed changes made.

**Comment by lrsmith** [comment]
Should be resolved with todays 0.2.0 release
